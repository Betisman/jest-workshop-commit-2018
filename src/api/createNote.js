
export default async (title, description) => {
  const response = await fetch('http://localhost:3001/note', {
    method: 'post',
    body: JSON.stringify({ title, description }),
    headers: { 'Content-Type': 'application/json' },
  });
  const notes = await response.json();
  return notes;
};