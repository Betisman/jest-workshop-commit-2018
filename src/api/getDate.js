
export default async () => {
  const response = await fetch('http://worldclockapi.com/api/json/est/now');
  const date = await response.json();
  return date;
};
