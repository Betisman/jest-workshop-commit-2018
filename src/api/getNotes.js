
export default async () => {
  const response = await fetch('http://localhost:3001/notes');
  const notes = await response.json();
  return notes;
};
