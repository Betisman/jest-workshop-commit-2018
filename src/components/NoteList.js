import React from 'react';
import ListItem from '../components/ListItem';

const NoteList = ({ items }) => (
  <div className="ui cards" style={{ justifyContent: 'center' }}>
    {items.map((obj) => (
      <ListItem key={obj.id} title={obj.title} description={obj.description} />
    ))}
  </div>
);

export default NoteList;
