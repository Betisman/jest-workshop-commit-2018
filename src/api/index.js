import getDate from './getDate';
import createNote from './createNote';
import getNotes from './getNotes';

export {
  getDate,
  createNote,
  getNotes,
};
