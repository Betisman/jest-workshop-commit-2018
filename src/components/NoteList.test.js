import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NoteList from './NoteList';
import { notes } from '../../test/fixtures/api';

const props = {
  items: notes,
};

describe('<NoteList />', function() {
  it('Match snapshot', () => {
    const tree = renderer.create(<NoteList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});