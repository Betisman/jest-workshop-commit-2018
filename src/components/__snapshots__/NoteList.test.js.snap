// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteList /> Match snapshot 1`] = `
<div
  className="ui cards"
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <div
    className="card"
  >
    <div
      className="content"
    >
      <div
        className="header"
        onClick={[Function]}
      >
        test title
      </div>
    </div>
  </div>
</div>
`;
